src/main/java/com/mongodb/jdbc/MongoDatabaseMetaData.java

```diff
@@ -406,19 +406,38 @@ public ResultSet getTableTypes() throws SQLException {
 
     // Helper for getting a stream of all database names.
     private Stream<String> getDatabaseNames() {
-        return this.conn.mongoClient.listDatabaseNames().into(new ArrayList<>()).stream();
+        return this.conn
+                .mongoClient
+                .listDatabaseNames()
+                .into(new ArrayList<>())
+                .stream()
+                .filter(dbName -> !dbName.isEmpty());
+    }
+
+    // Helper for getting a list of collection names from the db
+    // Using runCommand instead of listCollections as listCollections does not support authorizedCollections option
+    private ArrayList<Document> getCollectionsFromRunCommand(MongoDatabase db) {
+        Document listCollectionsResult =
+                db.runCommand(
+                        new Document("listCollections", 1)
+                                .append("authorizedCollections", true)
+                                .append("nameOnly", true));
+        ArrayList<Document> firstBatch =
+                (ArrayList<Document>)
+                        listCollectionsResult
+                                .get("cursor", Document.class)
+                                .getList("firstBatch", Document.class);
```

Comment:
It looks like the Java driver has a mechanism for getting a cursor on runCommand. While I think Nathan you asked and it seems the results should always be in 1 doc, I feel that if we can have cursor support, I'd feel much better. Especially when I read the doc for the listCollections command : https://www.mongodb.com/docs/manual/reference/command/listCollections/#mongodb-data-listCollections.cursor

Reply:
I tried getting the cursor mechanism to work, I don't think it will for listCollections since the cursor needs the caller to specify a collection name for the `getMore` command and listCollections isn't specific to a single collection.  I have an attempt at [this branch](https://github.com/bucaojit/mongo-jdbc-driver/tree/getMoreCursorAttempt), then I realized I was not setting the right collection name.